actionpack/lib/action_controller/metal/strong_parameters.rb

```diff
@@ -580,6 +580,24 @@ def stringify_keys # :nodoc:
       dup
     end
 
+    def method_missing(method_sym, *args, &block)
+      if @parameters.respond_to?(method_sym)
+        message = <<-DEPRECATE.squish
+          Method #{ method_sym } is deprecated and will be removed in Rails 5.1,
+          as `ActionController::Parameters` no longer inherits from
+          hash. Using this deprecated behavior exposes potential security
+          problems. If you continue to use this method you may be creating
+          a security vulunerability in your app that can be exploited. Instead,
+          consider using one of these documented methods which are not
+          deprecated: http://edgeapi.rubyonrails.org/classes/ActionController/Parameters.html
```

Comment:
Let's point to `api.rubyonrails.org` since when Rails 5.0 is released this will point to Rails 5.1 API.